src/controller/java/src/chip/devicecontroller/ChipTLVType.java

```diff
@@ -0,0 +1,366 @@
+package chip.devicecontroller;
+
+import java.util.ArrayList;
+
+public class ChipTLVType {
+  public enum TLVType {
+    UInt,
+    Int,
+    String,
+    ByteArray,
+    Boolean,
+    Float,
+    Double,
+    Null,
+    Struct,
+    Array,
+    Empty;
+  }
+
+  public static interface BaseTLVType {
+    TLVType type();
+
+    <T> T value(Class<T> clazz);
```

Comment:
is 'clazz' on purpose or typo?

Reply:
@yufengwangca 
"class" is a reserved word in Java and cannot be used as a variable.
So, when creating a Class' class object variable in Java, it is common to use “clazz”.